scanpy/plotting/_tools/__init__.py

```diff
@@ -633,3 +636,110 @@ def sim(adata, tmax_realization=None, as_heatmap=False, shuffle=False,
                    highlightsX=np.arange(tmax, n_realizations*tmax, tmax),
                    xlabel='index (arbitrary order)')
         utils.savefig_or_show('sim_shuffled', save=save, show=show)
+
+
+def embedding_density(
+        adata: AnnData,
+        basis: str,
+        key: str,
+        group: Optional[str] = None,
+        color_map: Union[Colormap, str] = 'YlOrRd',
+        greysize: Optional[int] = 20,
+        colorsize:  Optional[int] = 40,
+        vmax:  Optional[int] = 1,
+        vmin:  Optional[int] = 0,
+        save: Union[bool, str, None] = None,
+        **kwargs):
+    """Plot the density of cells in an embedding (per condition)
+
+    Plots the gaussian kernel density estimates (over condition) from the
+    `sc.tl.density()` output.
+
+    This function was written by Sophie Tritschler and implemented into
+    Scanpy by Malte Luecken.
+    
+    Parameters
+    ----------
+    adata : :class:`~anndata.AnnData`
+        The annotated data matrix.
+    basis : `str`
+        The embedding over which the density was calculated. This embedded
+        representation should be found in `adata.obsm['X_[basis]']``.
+    key : `str`
+        Name of the `.obs` covariate that contains the density estimates
+    group : `str`, optional (default: `None`)
+        Category of the observed covariate which will be plotted.
+    """
+    sanitize_anndata(adata)
+    
+    # Test user inputs
+    basis = basis.lower()
+
+    if basis == 'fa':
+        basis = 'draw_graph_fa'
+
+    if 'X_'+basis not in adata.obsm.dtype.names:
+        raise ValueError('Cannot find the embedded representation `adata.obsm[X_{!r}]`. Compute the embedding first.'.format(basis))
+
+    components = [0,1]
+
+    if basis == 'dm':
```

Comment:
I think this is `X_diffmap`, see https://github.com/theislab/scanpy/blob/master/scanpy/tools/_diffmap.py#L32.

Reply:
good spot :)